### DISCUSSION TITLE
DSC battery 0/1: Empty or OK?

### DISCUSSION THREAD

**Discussion by eigenphase** [merbanan/rtl_433#2195] [Q&A] [answered]
I have a bunch of DSC EW-DW4975 contact sensors. If I find this information:

```
time      : 1665169206.664511
model     : DSC-Security id        : 2117023
closed    : 1            event     : 0             tamper    : 0             Battery   : 1             xactivity : 0             xtamper1  : 0             xtamper2  : 0             exception : 0             esn       : 204d9f        status    : 195           status_hex: c3            Integrity : CRC
Modulation: ASK          Freq      : 433.9 MHz
RSSI      : -2.2 dB      SNR       : 19.3 dB       Noise     : -21.5 dB
```

What does Battery=1 mean? That the battery is GOOD or EMPTY?

When using mqtt, this translates to a field "battery_ok" from which I would infer battery=1 means GOOD. On the other hand, for a battery sensor in HomeAssistant, battery=1 means EMPTY, so I have to invert the bit in the config:

```
payload_on: 0
payload_off: 1
```

Interestingly I notice that sometimes devices go offline (become unavailable) even though Battery: 1 ... and sometimes they stay open even though they are closed (indicating that they did not send a message). So maybe battery=0 means GOOD and battery=1 means EMPTY?

Hence I want to confirm what the exact meaning of this bit is.

**Comment by zuckschwerdt** [comment, marked as answer]
All reported battery values are "battery_ok" with values 0 – 1. https://triq.org/rtl_433/DATA_FORMAT.html#common-device-data
Some devices report more fine grained steps. Best to map this to a percentage of "battery level", e.g. x100 for percentage full.

**Reply by eigenphase** [reply]
Thanks for confirming and the pointer. Apologize for not finding the docs (I have read the docs before)

So it seems inversion indeed needs to happen and sadly battery is not the reason why some devices do not work reliably...

**Comment by Redspray00** [comment]
My discover (rtl_433(Next)) log looks the same as yours but my MQTT broker doesn't want to display the status, "closed:". Would you mind sharing some of your configuration?
